cumulus/zombienet/zombienet-sdk/tests/zombie_ci/runtime_upgrade.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::anyhow;
+use std::time::Duration;
+
+use cumulus_zombienet_sdk_helpers::assert_para_throughput;
+
+use polkadot_primitives::Id as ParaId;
+use subxt::{OnlineClient, PolkadotConfig};
+use zombienet_configuration::types::AssetLocation;
+use zombienet_sdk::{
+	tx_helper::{ChainUpgrade, RuntimeUpgradeOptions},
+	LocalFileSystem, Network, NetworkConfigBuilder,
+};
+
+const PARA_ID: u32 = 2000;
+const WASM_WITH_SPEC_VERSION_INCREMENTED: &str =
+	"/tmp/wasm_binary_spec_version_incremented.rs.compact.compressed.wasm";
+
+async fn wait_for_upgrade(
+	client: OnlineClient<PolkadotConfig>,
+	expected_version: u32,
+) -> Result<(), anyhow::Error> {
+	let updater = client.updater();
+	let mut update_stream = updater.runtime_updates().await?;
+
+	while let Some(Ok(update)) = update_stream.next().await {
+		let version = update.runtime_version().spec_version;
+		log::info!("Update runtime spec version {version}");
+		if version == expected_version {
+			break;
+		}
+	}
+	Ok(())
+}
+// This tests makes sure that it is possible to upgrade parachain's runtime
+// and parachain produces blocks after such upgrade.
+#[tokio::test(flavor = "multi_thread")]
+async fn runtime_upgrade() -> Result<(), anyhow::Error> {
+	let _ = env_logger::try_init_from_env(
+		env_logger::Env::default().filter_or(env_logger::DEFAULT_FILTER_ENV, "info"),
+	);
+
+	log::info!("Spawning network");
+	let network = initialize_network().await?;
+
+	let alice = network.get_node("alice")?;
+	let alice_client: OnlineClient<PolkadotConfig> = alice.wait_client().await?;
+
+	log::info!("Ensuring parachain is registered");
+	assert_para_throughput(
+		&alice_client,
+		20,
+		[(ParaId::from(PARA_ID), 2..40)].into_iter().collect(),
+	)
+	.await?;
+
+	let timeout_secs: u64 = 250;
+	let charlie = network.get_node("charlie")?;
+	let charlie_client: OnlineClient<PolkadotConfig> = charlie.wait_client().await?;
+
+	let current_spec_version =
+		charlie_client.backend().current_runtime_version().await?.spec_version;
```

Comment:
dq: We are no longer validating block heights to make the test more stable wrt flaky runs?

Reply:
Assumed that waiting 20 blocks for registration is enough here since `charlie` is the only parachain node that produces blocks, so no need to validate block height.